docs/cfg_state_pattern.txt

```diff
@@ -0,0 +1,10 @@
+This document contains a context free grammar for the state pattern created for the Use cases.
+Reading this, one can form a map of how the states progress among ech other.
+
+=====================================================================================================
+
+CanvasIdleState --> CanvasIdleState | DownOnEmptyState | EditLinkState | AddToSelectionState
+DownOnEmptyState --> CanvasIdleState | EditLinkState
+EditLinkState --> CanvasIdleState | EditLinkState
+AddToSelectionState --> CanvasIdleState | MoveSelectionState
+MoveSelectionState --> CanvasIdleState | MoveSelectionState
```

Comment:
This is incomplete.  It is missing events, tests and reactions.  Typically each rule will be of the form
  State -->  Event Test Reaction State
However for transitory states we might not have the event.